### ISSUE TITLE
[Question]: Trying to run accessibility-checker in an AWS Lambda. Need to specify the path for saving the engine or disable loading the engine

### ISSUE THREAD

**Issue by mannika19** [IBMa/equal-access#1131]
### Project

a11y checker

### Description

Hi, I am trying to run accessibility-checker on an AWS Lambda. I am using a Puppeteer page. The accessibility-checker tries to create an `engine` folder  but AWS Lambda doesn't allow it to create a folder in the `var/task/` folder. I get this error:
```
Error: EROFS: read-only file system, mkdir '/var/task/accessibilityscanningservice/engine'
    at Object.mkdirSync (fs.js:1014:3)
    at Function.<anonymous> (/var/task/accessibilityscanningservice/index.js:649681:22)
    at step (/var/task/accessibilityscanningservice/index.js:649550:23)
    at Object.next (/var/task/accessibilityscanningservice/index.js:649497:18)
    at fulfilled (/var/task/accessibilityscanningservice/index.js:649468:28)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  errno: -30,
  syscall: 'mkdir',
  code: 'EROFS',
  path: '/var/task/accessibilityscanningservice/engine'
}
```
/var/task is where the Lambda function code is located, and in the actual Lambda environment, that filesystem is read-only. If you need to write to a file, you need to write to `/tmp`
Is there a way to use accessibility-checker (for node js) in an AWS lambda function?

**Comment by shunguoy** [comment]
@mannika19  if you use Puppeteer, have you tried the example step [here](https://github.com/IBMa/equal-access/tree/master/accessibility-checker-engine#programmatic)?  You should get a report object, rather than a report stored in a local directory.

**Comment by tombrunet** [comment]
@shunguoy This is regarding where the engine file is saved to for loading locally. The checker assumes the node_modules directory is writable. We may need a config parameter for what to use as a cache directory for that.

**Comment by mannika19** [comment]
Hi,

Is the config parameter for a cache directory something that is supported
today? If so, how and where do I define the parameter?

Sincerely,
Mannika Kshettry.
Pronouns: she, her, hers

**Comment by mannika19** [comment]
Hi @tombrunet, I wanted to follow up on my previous message

**Comment by tombrunet** [comment]
Will track this in this PR: https://github.com/IBMa/equal-access/pull/1138

**Comment by mannika19** [comment]
@aliunwala Thank you! Would the command-line switch work with the `accessibility-checker` package as well? Our implementation needs to be able to run IBM checker via code.

**Comment by mannika19** [comment]
Hi, Can we get an ETA on when this fix will be pushed?

**Comment by tombrunet** [comment]
This will show up in our next release. Checking on what that will be. The default location was moved to the OS temp directory with an ability to override using the `cacheFolder` config property.

**Comment by eirckr** [comment]
@mannika19 We hope to get this out today in version 3.1.41

**Comment by tombrunet** [comment]
@mannika19 This was released last night. Let us know whether or not it's working for you now.

**Comment by mannika19** [comment]
@tombrunet Hi, I am trying to run accessibility-checker on my AWS lambda. I get this error:
```
TypeError: Cannot convert undefined or null to object
    at t.clearCaches (/tmp/accessibility-checker/engine/ace-node.js:2:466503)
    at e.run (/tmp/accessibility-checker/engine/ace-node.js:2:176251)
    at e.check (/tmp/accessibility-checker/engine/ace-node.js:2:194009)
    at /var/task/accessibilityscanningservice/index.js:652079:36
    at step (/var/task/accessibilityscanningservice/index.js:651665:23)
    at Object.next (/var/task/accessibilityscanningservice/index.js:651612:18)
    at /var/task/accessibilityscanningservice/index.js:651598:71
    at new Promise (<anonymous>)
    at __awaiter (/var/task/accessibilityscanningservice/index.js:651580:14)
    at getComplianceHelperLocal (/var/task/accessibilityscanningservice/index.js:652068:14)
```
I found issue #171  where they had a similar error with the command line argument. I can't view the `/tmp/` folder for the lambda as it gets disposed of after the lambda execution. I can't find a way for me to debug the error in the `clearCaches` function. Would you know what might be going on here?
